expat/configure.ac

```diff
@@ -183,33 +183,75 @@ AC_LINK_IFELSE([AC_LANG_SOURCE([
       AC_MSG_RESULT([yes])],
      [AC_MSG_RESULT([no])])])
 
-
-AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
-AC_LINK_IFELSE([AC_LANG_SOURCE([
-  #include <stdlib.h>  /* for NULL */
-  #include <sys/random.h>
-  int main() {
-    return getrandom(NULL, 0U, 0U);
-  }
-  ])],
-  [AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
-   AC_MSG_RESULT([yes])],
-  [AC_MSG_RESULT([no])
-
-   AC_MSG_CHECKING([for syscall SYS_getrandom (Linux 3.17+)])
+AC_ARG_WITH([usegetrandom],
+ [AS_HELP_STRING([--with-usegetrandom],
+ [Use getrandom function in the system @<:@default=check@:>@])],
+ [],
+ [with_usegetrandom=check])
+          
+AS_IF([test "x$with_usegetrandom" != xno],
+  [AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
+   AC_LINK_IFELSE([AC_LANG_SOURCE([
+   #include <stdlib.h>  /* for NULL */
+   #include <sys/random.h>
+   int main() {
+     return getrandom(NULL, 0U, 0U);
+   }
+   ])],
+   [
+	AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
+    AC_MSG_RESULT([yes])
+    getrandom_detect=yes
+   ],
+    [AC_MSG_RESULT([no])]
+     getrandom_detect=no)
+	],
+   [if test "x$with_usegetrandom" != xcheck; then
+       AC_DEFINE([HAVE_GETRANDOM], [0], [Define to 0, dont use the `getrandom' function.])
```

Comment:
From https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/html_node/Defining-Symbols.html:
> If multiple `AC_DEFINE` statements are executed for the same variable name (not counting any parenthesized argument list), the last one wins.

So please drop the second call to `AC_DEFINE` to get back the right description text for `HAVE_GETRANDOM`. Yes, `AC_DEFINE` is weird :smile: